fedimint-api/src/db/mod.rs

```diff
@@ -247,6 +261,77 @@ impl Drop for CommitTracker {
     }
 }
 
+/// ModuleDatabaseTransaction is a wrapper around IsolatedDatabaseTransaction that
```

Comment:
Why do we need ... a wrapper around a wrapper? Could this be just 

```
struct ModuleDatabaseTransaction<'a> {
    dbtx: DatabaseTransaction<'a>,
}
```

where `dbtx` is already wrapped?

Reply:
I'm actually running into problems when I do that, because the `with_module_prefix` function takes a mutable reference to self, so it is still borrowed when trying to return the new ModuleDatabaseTransaction (that has a wrapped dbtx).

cannot return value referencing function parameter `dbtx`
returns a value referencing data owned by the current function

Let me know if I'm missing something but the current way seems the easiest to me?

Reply:
It should work - in similar way that it works for `IsolatedDatabaseTransaction` to wrap the inner transaction.

Possibly here you need `dbtx: &mut dyn IDatabaseTransaction<'a>` as well - same way?

Reply:
I don't think that works either, otherwise I could just reuse IsolatedDatabaseTransaction and I wouldn't need another wrapper. I believe the issue is that I need to create a DatabaseTransaction in begin_transaction that is wrapped, from nothing. I can't borrow a DatabaseTransaction that was created in begin_transaction and then return the new wrapped one, because the new one still has a reference to the old one that has now dropped out of scope.